### PR TITLE
add graph_compact.

### PR DESCRIPTION
**Pull request by zheng-da**
## Description ##
dgl_graph_compact works with `csr_neighbor_uniform_sample` and `csr_neighbor_non_uniform_sample`. These two sampling operators create CSR matrices with many empty rows. This operator compacts these CSR matrices.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by vandanavk** [comment]
@mxnet-label-bot add [Operator, pr-awaiting-review]

File: src/operator/contrib/dgl_graph.cc [hunk 1]
```diff
@@ -1390,5 +1390,196 @@ Example::
...
+  }
+  for (size_t i = 0; i < out_attrs->size(); i++) {
+    out_attrs->at(i) = mshadow::kInt64;
+  }
+  return true;
+}
+
+NNVM_REGISTER_OP(_contrib_dgl_graph_compact)
+.describe(R"code(This operator compacts a CSR matrix generated by
+csr_neighbor_uniform_sample and csr_neighbor_non_uniform_sample.
```
Comment (eric-haibin-lin) [inline src/operator/contrib/dgl_graph.cc]: I think the name csr_neighbor_uniform_sample has been changed

File: src/operator/contrib/dgl_graph.cc [hunk 2]
```diff
@@ -1390,5 +1390,196 @@ Example::
...
+The CSR matrices generated by these two operators may have many empty
+rows at the end. This operator removes these empty rows and empty columns.
+Example::
+  subgs = mx.nd.contrib.csr_neighbor_uniform_sample(csr, seed, num_hops=1,
+          num_neighbor=2, max_num_vertices=5)
+  subg_v = subgs[0]
+  subg = subgs[1]
+  compacts = mx.nd.contrib.dgl_graph_compact(subg, subg_v,
+          graph_sizes=(subg_v[-1].asnumpy()[0]))
```
Comment (eric-haibin-lin) [inline src/operator/contrib/dgl_graph.cc]: Can you also print the input/output if dgl_graph_compact?

File: src/operator/contrib/dgl_graph.cc [hunk 3]
```diff
@@ -1390,5 +1390,196 @@ Example::
...
+                            const NDArray &out_csr, size_t graph_size) {
+  TBlob in_idx_data = csr.aux_data(csr::kIdx);
+  TBlob in_ptr_data = csr.aux_data(csr::kIndPtr);
+  const dgl_id_t *indices_in = in_idx_data.dptr<dgl_id_t>();
+  const dgl_id_t *indptr_in = in_ptr_data.dptr<dgl_id_t>();
+  const dgl_id_t *row_ids = vids.data().dptr<dgl_id_t>();
+  size_t num_elems = csr.aux_data(csr::kIdx).shape_.Size();
+  // The last element in vids is the actual number of vertices in the subgraph.
+  CHECK_EQ(vids.shape()[0], in_ptr_data.shape_[0]);
+  CHECK_EQ((size_t) row_ids[vids.shape()[0] - 1], graph_size);
```
Comment (eric-haibin-lin) [inline src/operator/contrib/dgl_graph.cc]: nit nit: use static_cast

**Comment by eric-haibin-lin** [comment]
Please follow doc style in https://github.com/apache/incubator-mxnet/pull/13463

**Comment by aaronmarkham** [comment]
Hi guys, this PR breaks the docs. Specifically, I think I need to remove these from the [Symbol Contrib API docs page](https://mxnet.incubator.apache.org/api/python/symbol/contrib.html):
isinf
isfinite  
isnan
They don't correlate to any function in Symbol. NDArray has something for these though. Any issue with me submitting a PR to remove those lines?

**Comment by eric-haibin-lin** [comment]
@aaronmarkham good catch. No concern from my side. Thanks for pointing that out!

**Pull request merged**